vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,180 @@
+    def __enter__(self):
+        super(MozillaDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, subdir="announce"
+            )
+
+    def updated_advisories(self) -> Set[Advisory]:
+        files = self._updated_files.union(self._added_files)
+        files = [
+            f for f in files if f.endswith(".md") or f.endswith(".yml")
+        ]  # skip irrelevant files
+
+        advisories = []
+        for path in files:
+            advisories.extend(to_advisories(path))
+
+        return self.batch_advisories(advisories)
+
+
+def to_advisories(path: str) -> List[Advisory]:
+    """
+    Convert a file to corresponding advisories.
+    This calls proper method to handle yml/md files.
+    """
+    mfsa_id = mfsa_id_from_filename(path)
+    if not mfsa_id:
+        return []
+
+    with open(path) as lines:
+        if path.endswith(".md"):
+            return get_advisories_from_md(mfsa_id, lines)
+        if path.endswith(".yml"):
+            return get_advisories_from_yml(mfsa_id, lines)
+
+    return []
+
+
+def get_advisories_from_yml(mfsa_id, lines) -> List[Advisory]:
+    advisories = []
+    data = yaml.safe_load(lines)
+    data["mfsa_id"] = mfsa_id
+
+    fixed_package_urls = get_package_urls(data.get("fixed_in"))
+    references = get_yml_references(data)
+
+    if not data.get("advisories"):
+        return []
+
+    for cve, advisory in data["advisories"].items():
+        # These may contain HTML tags
+        summary = BeautifulSoup(advisory.get("description", ""), features="lxml").get_text()
+
+        advisories.append(
+            Advisory(
+                summary=summary,
+                vulnerability_id=cve if is_cve(cve) else "",
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        )
+
+    return advisories
+
+
+def get_advisories_from_md(mfsa_id, lines) -> List[Advisory]:
+    yamltext, mdtext = split_markdown_front_matter(lines.read())
+    data = yaml.safe_load(yamltext)
+    data["mfsa_id"] = mfsa_id
+
+    fixed_package_urls = get_package_urls(data.get("fixed_in"))
+    references = get_yml_references(data)
+    cves = re.findall(r"CVE-\d+-\d+", yamltext + mdtext, re.IGNORECASE)
+    for cve in cves:
+        references.append(
+            Reference(
+                reference_id=cve,
+                url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve}",
+            )
+        )
+
+    description = html_get_p_under_h3(markdown(mdtext), "description")
+
+    return [
+        Advisory(
+            summary=description,
+            vulnerability_id="",
+            impacted_package_urls=[],
+            resolved_package_urls=fixed_package_urls,
+            references=references,
+        )
+    ]
+
+
+def html_get_p_under_h3(html, h3: str):
```

Comment:
Find a better name and add a docstring. The name should be descriptive about what it returns or does, not how it does it. Here `html_get_p_under_h3` tells me nothing at all.
Think of it this way: 3 months from now, you are reading this code again. the function and docstring should make this obvious and self-evident.

Reply:
Updated
```py
def html_get_section(html, heading, heading_tag="h3"):
    """
    Return the contents of all <p> tags below the provided header (by default h3 tag)
    eg:
    <h3>Description</h3>
    <p> Here goes the desc.</p>
    <p> More desc.</p>

    html_get_section("description") would return "Here goes the desc.More desc"
    """
    soup = BeautifulSoup(html, features="lxml")
    h3tag = soup.find(heading_tag, text=lambda txt: txt.lower() == heading.lower())
```

Does it look better?

Reply:
Better but why having `heading_tag="h3"` as an argument?
Also if you collect paragraphs, then you should join them with a line return and glued not as in `"Here goes the desc.More desc"`

While you are at it, using a doctest in the docstring works too since this is mostly what you are doing already here

Reply:
Also since this is only used to get a description, why not call this `get_description(html_text)` and hide "how the sausage is made" inside the function?

Reply:
Done